### PR TITLE
Feature fish animation

### LINKED ISSUE #36: Fish Animation with Groupings

### LINKED ISSUE #41: Add and fix unit tests for fish animation

### PR DESCRIPTION
**Pull request by bingxiaan**
bella: added new animation method that uses images and is dependent on score
- closes #36, #37, #38

bingxian: added tests, fixed animation issue when bar is full
- closes #41, #42

### CODE REVIEW

File: app/src/main/java/com/hydrofish/app/ui/composables/tabs/HomeScreen.kt [hunk 1]
```diff
@@ -62,19 +72,244 @@ fun HomeScreen(modifier: Modifier = Modifier, hydroFishViewModel: HydroFishViewM
...
-    Box(modifier = modifier
-        .background(largeRadialGradient),
+    Box(
+        modifier = modifier.background(largeRadialGradient),
         contentAlignment = Alignment.Center,
-
-        ) {
-        DisplayFish(modifier = Modifier, fishes = hydroFishUIState.fishTypeList, distances = hydroFishUIState.fishDistances)
+    ) {
+        /*hydroFishViewModel.getAllFish()*/
```
Comment (isabellachen3988) [inline app/src/main/java/com/hydrofish/app/ui/composables/tabs/HomeScreen.kt]: remember to remove this

File: app/src/main/java/com/hydrofish/app/ui/composables/tabs/HomeScreen.kt [hunk 2]
```diff
@@ -62,19 +72,244 @@ fun HomeScreen(modifier: Modifier = Modifier, hydroFishViewModel: HydroFishViewM
...
+                animatableMap[AnimatableType.X]?.animateTo(
+                    targetValue = -400f,
+                    animationSpec = tween(
+                        durationMillis = 3000,
+                        easing = EaseInOutElastic
+                    )
+                )
+            }
+
+            /*launch {
```
Comment (isabellachen3988) [inline app/src/main/java/com/hydrofish/app/ui/composables/tabs/HomeScreen.kt]: is there a reason you left this in?

File: app/src/main/java/com/hydrofish/app/ui/HydroFishUIState.kt [hunk 3]
```diff
@@ -1,11 +1,26 @@
...
+    val animationGroupPositionHandler: AnimationGroupPositionHandler
+        = AnimationGroupPositionHandler(
+            listOf(
+                hashSetOf(AnimatableType.X, AnimatableType.FLIP),
+                //*hashSetOf(AnimatableType.CAMERASCALE)*//*
+                //*hashSetOf(AnimatableType.SCALE),*//*
+                hashSetOf(AnimatableType.Y, AnimatableType.ROTATE),
+                hashSetOf(AnimatableType.DIAGONAL_X, AnimatableType.DIAGONAL_Y, AnimatableType.DIAG_FLIP),
+                hashSetOf(AnimatableType.DIAGONAL_X_R, AnimatableType.DIAGONAL_Y_R, AnimatableType.DIAG_FLIP_R),
+                /*hashSetOf(AnimatableType.X, AnimatableType.ROTATE),*/
```
Comment (isabellachen3988) [inline app/src/main/java/com/hydrofish/app/ui/HydroFishUIState.kt]: is there a reason you left these in?

File: app/src/main/java/com/hydrofish/app/ui/composables/tabs/HomeScreen.kt (hunk 2 above)
Comment (isabellachen3988) [reply app/src/main/java/com/hydrofish/app/ui/composables/tabs/HomeScreen.kt]: please check code that you commented out. if you intended to leave the code there, write a comment that specifies why you're leaving it in

File: app/src/main/java/com/hydrofish/app/HydroFishViewModel.kt [hunk 4]
```diff
@@ -12,34 +12,37 @@ class HydroFishViewModel: ViewModel() {
...
-                    dailyWaterConsumedML = amt + currentState.dailyWaterConsumedML,
+                    fishScore = currentState.fishScore + 1
+
                 )
             }
+            uiState.value.animationGroupPositionHandler.prepForPopulation()
+
+
+        }
+        else {
```
Comment (isabellachen3988) [inline app/src/main/java/com/hydrofish/app/HydroFishViewModel.kt:34]: empty else statement

File: app/src/test/java/com/hydrofish/app/HomepageUnitTest.kt [hunk 5]
```diff
@@ -20,17 +20,18 @@ class HomepageUnitTest {
...
-        val newFishCount = viewModel.uiState.value.fishTypeList.count();
-        assert(newFishCount == initialFishCount + 1);
-
-        // once we exceed the limit, no additional fish should be added
-        viewModel.increaseWaterLevel(1);
-        assert(newFishCount == viewModel.uiState.value.fishTypeList.count());
-    }
+    // test no longer works - require a new test
+//    @Test
+//    fun testFishAddition() {
```
Comment (isabellachen3988) [inline app/src/test/java/com/hydrofish/app/HomepageUnitTest.kt]: will you be rewriting a test for this?

File: app/src/main/java/com/hydrofish/app/ui/composables/tabs/HomeScreen.kt (hunk 1 above)
Comment (bingxiaan) [reply app/src/main/java/com/hydrofish/app/ui/composables/tabs/HomeScreen.kt]: Removed extra things.

File: app/src/main/java/com/hydrofish/app/ui/composables/tabs/HomeScreen.kt (hunk 2 above)
Comment (bingxiaan) [reply app/src/main/java/com/hydrofish/app/ui/composables/tabs/HomeScreen.kt]: Removed extra things.

File: app/src/main/java/com/hydrofish/app/ui/HydroFishUIState.kt (hunk 3 above)
Comment (bingxiaan) [reply app/src/main/java/com/hydrofish/app/ui/HydroFishUIState.kt]: Removed extra things.

**Review by isabellachen3988 — APPROVED** [review]
all good

**Pull request closed without merging**